crates/polars-parquet/src/arrow/read/deserialize/binview.rs

```diff
@@ -626,11 +635,20 @@ impl utils::Decoder for BinViewDecoder {
             },
         }
 
+        let buffer = &buffer[..buffer.len() - page_values.values.len()];
+
         if self.check_utf8.load(Ordering::Relaxed) {
-            // @TODO: Better error message
-            values
-                .validate_utf8(buffer_offset, views_offset)
-                .map_err(|_| ParquetError::oos("Binary view contained invalid UTF-8"))?
+            // This is a small trick that allows us to check the Parquet buffer instead of the view
+            // buffer. Batching the UTF-8 verification is more performant. For this to be allowed,
+            // all the interleaved lengths need to be valid UTF-8.
+            if max_length < 128 {
```

Comment:
Does this validate all strings concatenated? If so, I am not sure this is valid.

Reply:
Yes, it does. I am reasonably certain it is valid. Why wouldn't it be valid?

Reply:
for instance **pi** contains of two unicode chars.

[206 160] is valid pi.

But `[206 | 160]` with a word boundary after 206 is invalid utf8.

Reply:
We check that all the lengths are valid UTF-8 (`max_length < 128`). The first byte of a length is then always `0`, which would always follow any individual string. In UTF-8, `0` is never a valid byte beyond the first byte.

Reply:
Resolved async. :+1:

Reply:
I believe that 'first byte of a length' should be 'top bit of the first byte of the length', and in UTF-8 multi-byte characters can't have the top bit set to 0.